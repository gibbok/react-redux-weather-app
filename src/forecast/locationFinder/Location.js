// @flow
import * as React from 'react'

/* eslint-disable no-undef */
type PropsType = {
  onLocationClick: Function,
  id: number,
  name: string,
  country:string
}
/* eslint-enable no-undef */

const Location:React.StatelessFunctionalComponent<any> = ({ onLocationClick, id, name, country }:PropsType):React.Element<any> => (
  <li>
    <a onClick={onLocationClick}>{name}, {country}</a>
  </li>
)

export default Location
