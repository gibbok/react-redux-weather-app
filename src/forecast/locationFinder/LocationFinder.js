// @flow
import * as React from 'react'
import Location from './Location'
import { withStyles, Menu } from 'material-ui'

/* eslint-disable no-undef */
type PropsType = {
  +locations: Array<Object>,
  +onLocationClick: Function,
  +isFinderOpen: boolean,
  +classes: Object
}
/* eslint-enable no-undef */

const styles = theme => {
  return ({
    root: {
    }
  })
}

const LocationFinder:React.StatelessFunctionalComponent<any> = ({ locations, onLocationClick, isFinderOpen, classes }:PropsType) => {
  const locationItems = locations.map((loc, index) =>
    <Location
      key={index}
      {...locations[index]}
      onLocationClick={() => onLocationClick(loc)}
      />
      )
  const locationList = locations.length > 0 ? <ul>{locationItems}</ul> : null
  const canOpen = locations.length > 0
  return (
    <Menu open={canOpen}>
      {locationList}
    </Menu>
    // <Menu open={isFinderOpen}>
    //   {locationList}
    // </Menu>
  )
}

export default withStyles(styles)(LocationFinder)

// const LocationFinder:React.StatelessFunctionalComponent<any> = ({ locations, onLocationClick }:PropsType) => {
//   const locationItems = locations.map((location, index) =>
//     <Location
//       key={index}
//       {...locations[index]}
//       onLocationClick={() => onLocationClick(location)}
//       />
//       )
//   const locationList = locations.length > 0 ? <ul>{locationItems}</ul> : null
//   return (locationList)
// }

// export default LocationFinder
