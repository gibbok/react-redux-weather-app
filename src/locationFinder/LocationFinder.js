import React from 'react'
import Location from './Location'

const LocationFinder = ({ locations, onLocationClick }) => (
  <ul>
    {locations.map((location, index) =>
      <Location
        key={index}
        {...locations[index]}
        onClick={() => onLocationClick(location)}
      />
    )}
  </ul>
)

export default LocationFinder
