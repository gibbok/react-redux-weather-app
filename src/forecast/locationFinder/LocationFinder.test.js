import React from 'react'
import { shallow } from 'enzyme'
import LocationFinder from './LocationFinder'
import Location from './Location'
import toJson from 'enzyme-to-json'

describe('<LocationFinder />', () => {
  it('should render two <Location /> components', () => {
    const data = [
      {
        country: 'CZ',
        id: 3067696,
        name: 'Prague'
      },
      {
        country: 'US',
        id: 4548393,
        name: 'Prague'
      }
    ]
    const wrapper = shallow(<LocationFinder locations={data} />)
    expect(wrapper.find(Location).length).toBe(2)
  })

  it('should render', () => {
    const data = [
      {
        country: 'CZ',
        id: 3067696,
        name: 'Prague'
      },
      {
        country: 'US',
        id: 4548393,
        name: 'Prague'
      }
    ]
    const wrapper = shallow(
      <LocationFinder locations={data} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
