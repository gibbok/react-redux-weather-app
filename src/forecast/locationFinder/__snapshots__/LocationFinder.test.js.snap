// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationFinder /> should render 1`] = `
<ul>
  <Location
    country="CZ"
    id={3067696}
    name="Prague"
    onLocationClick={[Function]}
  />
  <Location
    country="US"
    id={4548393}
    name="Prague"
    onLocationClick={[Function]}
  />
</ul>
`;
